### ISSUE TITLE
.removeScript does not remove NPM script

### ISSUE THREAD

**Issue by zARODz11z** [projen/projen#631]
Hello I am working on converting an existing AWS CDK Construct Library to use Projen however I would like to remove some of the auto generated npm scripts. For example I created a AwsCdkConstructLibrary using Projen and some scripts were auto generated like bump, release, compat, and so on but I already have existing shell scripts to do those actions and favor my scripts at the moment (in the future I may use the generated scripts). I see the API has a .removeScript function however after synthesizing, the scripts will not be removed. I could get around this with the escape hatches but that means I would get rid of it in the tasks.json and package.json manually which will clutter my projenrc file. 

To summarize I am trying to use the .removeScript function on an  AwsCdkConstructLibrary created by projen but the scripts will not be removed as I believe the function should. Based off of this https://github.com/projen/projen/blob/main/docs/awscdk-construct.md it seems that an AwsCdkConstructLibrary is a NodeProject so im not sure why the .removeScript function is not working. I am using yarn as the default to run the scripts, although im not sure if that has anything to do with it.

**Comment by zARODz11z** [comment]
@Chriscbr  Thank you for working on this, I have a question if you choose to remove a NPM script via the `.removeScript` function then what is the purpose of still having it in Projen's `tasks.json` file? I was under the impression that the `removeScript` function would remove the script from the package.json and tasks.json files Is there a way of removing a task without having to use an escape hatch manually?
Thanks in advance!

**Comment by Chriscbr** [comment]
@zARODz11z Good question. The task still exists so that it can be used in other contexts, like for GitHub actions workflows. projen's logic is currently set up so that task information can be used in many synthesized files, while setScript/removeScript only affects package.json.

Right now removing tasks isn't easy _in part_ because we've modeled `Project` type interface to always include certain tasks, e.g. every `NodeProject` has a build task accessible at `project.buildTask`.

projen's declarative style makes it easy to add things, but removing things is a bit trickier...

Q: Would resetting the tasks (and maybe changing the commands that are run) work in your situation? For example:

```ts
project.buildTask.reset();
project.tasks.tryFind('compat').reset();
```

If not, I have a couple ideas for library changes that could address this:

- make tasks removable with a standard API, and somehow enforce that task-based fields are optional (I feel like this isn't great UX, because in your code you could still have a variable referencing a task, even though it has been "destroyed")
- add some option(s) to omit creating these default tasks somehow?
- add an new field and corresponding API for tasks that "hides" them from the CLI and other default script listings such as package.json

**Comment by zARODz11z** [comment]
@Chriscbr That makes sense, thank you for the explanation! I was working on converting an AWS CDK Construct Library to use Projen but the issue was that our library already had an existing release workflow (that would also version bump) so it did not make sense to keep some of the auto generated Projen scripts (release, bump).

We wanted the only way to release our CDK Construct library to be through a custom script so we opted to remove the npm scripts and projen tasks in efforts of keeping the release workflow to just 1 script that we created. Having the ability to remove tasks would be nice in our efforts of not confusing our existing release workflow and keeping our projenrc file clean. However for the moment we can work around it.
